### PR TITLE
Dev v2

### PR DESCRIPTION
**Pull request by anil-plivo**

### CODE REVIEW

**Comment by nalin-plivo** [comment]
we should not be doing merges into master branch unless releasing for GA

File: PlivoAddressBook/app/src/main/java/com/plivo/plivoaddressbook/BaseFragment.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        // dummy view for appium test
```
Comment (Altanaib) [inline PlivoAddressBook/app/src/main/java/com/plivo/plivoaddressbook/BaseFragment.java:31]: fix comment . Ambiguous

File: PlivoAddressBook/app/src/main/java/com/plivo/plivoaddressbook/BaseFragment.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        }
+    }
+
+    protected void removeFragment() {
+        getChildFragmentManager().beginTransaction()
+                .remove(this)
+                .commitAllowingStateLoss();
+    }
+
+    public void updateUi(Call call) {
```
Comment (Altanaib) [inline PlivoAddressBook/app/src/main/java/com/plivo/plivoaddressbook/BaseFragment.java:62]: empty function ?

File: PlivoAddressBook/app/src/main/java/com/plivo/plivoaddressbook/layer/impl/PlivoSDKImpl.java [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+    public PlivoSDKImpl(PlivoCallStack callObj, ContactUtils contactUtils) {
+        super(callObj, contactUtils);
+    }
+
+    public boolean isLoggedIn() {
+        return endpoint().getRegistered();
+    }
+
+    public void keepAlive(PlivoBackendListener.LoginListener listener) {
+        // todo: enable keepAlive / loginAgain, once fixed on SDK
```
Comment (Altanaib) [inline PlivoAddressBook/app/src/main/java/com/plivo/plivoaddressbook/layer/impl/PlivoSDKImpl.java]: ambiguous comment for new developers who use plivo android SDK to get started

File: PlivoAddressBook/app/src/main/java/com/plivo/plivoaddressbook/layer/impl/PlivoSDKImpl.java [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+
+    private Incoming incoming() {
+        return (Incoming) getCurrentCall().getData();
+    }
+
+    private Outgoing outgoing() {
+        return (Outgoing) getCurrentCall().getData();
+    }
+
+    private String from(String fromContact, String fromSip) {
```
Comment (Altanaib) [inline PlivoAddressBook/app/src/main/java/com/plivo/plivoaddressbook/layer/impl/PlivoSDKImpl.java:144]: add description comment

File: PlivoAddressBook/app/src/main/java/com/plivo/plivoaddressbook/layer/plivo/PlivoBackend.java [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        if (callStack.removeFromCallStack(c)) {
+            notifyCallStackChange(c);
+        }
+    }
+
+    public void clearCallStack() {
+        callStack.clearCallStack();
+    }
+
+    // Exceptions
```
Comment (Altanaib) [inline PlivoAddressBook/app/src/main/java/com/plivo/plivoaddressbook/layer/plivo/PlivoBackend.java:171]: ?

File: PlivoAddressBook/app/src/main/java/com/plivo/plivoaddressbook/screens/dial/calls/OngoingCallFragment.java [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        ButterKnife.bind(this, view);
+
+        viewModel.incomingDTMFObserver().observe(this, digit -> dtmfTextView.setText(dtmfTextView.getText() + " " + digit));
+        dialer.setOnDigitClickListener(digit -> viewModel.sendDTMF(digit));
+        tickManager.setTickListener(callTimer);
```
Comment (Altanaib) [inline PlivoAddressBook/app/src/main/java/com/plivo/plivoaddressbook/screens/dial/calls/OngoingCallFragment.java:92]: suggestion - on inactive callstate , dont show timer

**Pull request closed without merging**